typespec-extension/src/main/java/com/azure/autorest/fluent/TypeSpecFluentPlugin.java

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.autorest.fluent;
+
+import com.azure.autorest.JavaSettingsAccessor;
+import com.azure.autorest.TypeSpecPlugin;
+import com.azure.autorest.extension.base.model.Message;
+import com.azure.autorest.extension.base.model.codemodel.CodeModel;
+import com.azure.autorest.extension.base.plugin.JavaSettings;
+import com.azure.autorest.fluent.mapper.FluentMapper;
+import com.azure.autorest.fluent.model.javamodel.FluentJavaPackage;
+import com.azure.autorest.fluentnamer.FluentNamer;
+import com.azure.autorest.mapper.Mappers;
+import com.azure.autorest.model.clientmodel.Client;
+import com.azure.core.util.CoreUtils;
+import com.azure.typespec.model.EmitterOptions;
+import com.azure.typespec.util.FileUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TypeSpecFluentPlugin extends FluentGen {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TypeSpecFluentPlugin.class);
+    private final EmitterOptions emitterOptions;
+
+    public TypeSpecFluentPlugin(EmitterOptions emitterOptions, boolean sdkIntegration) {
+        super(new TypeSpecPlugin.MockConnection(), "dummy", "dummy");
+
+        this.emitterOptions = emitterOptions;
+        SETTINGS_MAP.put("namespace", emitterOptions.getNamespace());
+        if (!CoreUtils.isNullOrEmpty(emitterOptions.getOutputDir())) {
+            SETTINGS_MAP.put("output-folder", emitterOptions.getOutputDir());
+        }
+        if (!CoreUtils.isNullOrEmpty(emitterOptions.getServiceName())) {
+            SETTINGS_MAP.put("service-name", emitterOptions.getServiceName());
+        }
+        if (emitterOptions.getGenerateSamples() != null) {
+            SETTINGS_MAP.put("generate-samples", emitterOptions.getGenerateSamples());
+        }
+        if (emitterOptions.getGenerateTests() != null) {
+            SETTINGS_MAP.put("generate-tests", emitterOptions.getGenerateTests());
+        }
+        if (emitterOptions.getFluent() != null) {
+            SETTINGS_MAP.put("fluent", emitterOptions.getFluent());
+        }
+        SETTINGS_MAP.put("sdk-integration", sdkIntegration);
+
+        JavaSettingsAccessor.setHost(this);
+        LOGGER.info("Output folder: {}", emitterOptions.getOutputDir());
+        LOGGER.info("Namespace: {}", JavaSettings.getInstance().getPackage());
+    }
+
+    public Client processClient(CodeModel codeModel) {
+        // transform code model
+        FluentNamer fluentNamer = new TypeSpecFluentNamer();
+        codeModel = fluentNamer.transform(codeModel);
```

Comment:
Personally I'd prefer Plugin sets the SETTINGS_MAP into FluentNamer, in ctor or other place, instead of FluentNamer pulling it out from Plugin.